src/main/java/org/folio/marc4ld/service/marc2ld/field/ResourceProcessorImpl.java

```diff
@@ -71,6 +71,13 @@ private void additionalMapping(Resource resource,
       .ifPresent(m -> m.map(new MarcData(dataField, controlFields), resource));
 
     resource.setId(hashService.hash(resource));
+    setLabel(resource);
+  }
+
+  private void setLabel(Resource resource) {
```

Comment:
At present, label and id is assigned at different places of the code. 
I am thinking to create a cleanup ticket to move label and ID assignment to the final step (ie, after creating all the resources and the edges). We can discuss tomorrow.